flink-runtime/src/main/java/org/apache/flink/runtime/resourcemanager/slotmanager/SlotManager.java

```diff
@@ -123,6 +125,9 @@
 	/** Release task executor only when each produced result partition is either consumed or failed. */
 	private final boolean waitResultConsumedBeforeRelease;
 
+	/** Resource profiles of slots that can be acquired. */
+	private final HashSet<ResourceProfile> allowedSlotResourceProfiles = new HashSet<>();
```

Comment:
I think the name is a little ambiguous and it seems to be the resource profiles of the slots that are allowed to created on the TM side. I think it may be better to use names like `availableSlotResourceProfiles`.